pkg/decode/decode_protobuf.go

```diff
@@ -67,26 +67,30 @@ func PBFlowToMap(flow *pbflow.Record) config.GenericMap {
 	}
 
 	ethType := ethernet.EtherType(flow.EthProtocol)
-	if ethType == ethernet.EtherTypeIPv4 || ethType == ethernet.EtherTypeIPv6 {
-		out["SrcAddr"] = ipToStr(flow.Network.GetSrcAddr())
-		out["DstAddr"] = ipToStr(flow.Network.GetDstAddr())
-		out["Proto"] = flow.Transport.GetProtocol()
-		out["Dscp"] = flow.Network.GetDscp()
-		proto := flow.Transport.GetProtocol()
-		if proto == syscall.IPPROTO_ICMP || proto == syscall.IPPROTO_ICMPV6 {
-			out["IcmpType"] = flow.GetIcmpType()
-			out["IcmpCode"] = flow.GetIcmpCode()
-		}
 
-		if proto == syscall.IPPROTO_TCP || proto == syscall.IPPROTO_UDP || proto == syscall.IPPROTO_SCTP {
-			if proto == syscall.IPPROTO_TCP {
-				out["SrcPort"] = flow.Transport.GetSrcPort()
-				out["DstPort"] = flow.Transport.GetDstPort()
-				out["Flags"] = flow.Flags
-			} else {
-				out["SrcPort"] = flow.Transport.GetSrcPort()
-				out["DstPort"] = flow.Transport.GetDstPort()
-			}
+	if ethType != ethernet.EtherTypeIPv4 && ethType != ethernet.EtherTypeIPv6 {
```

Comment:
@msherif1234 I'm doing this early return here when it's not ipv4/ipv6, could you confirm this is correct?
Previously, in that case (e.g. with ARP), the code would still check for drops/DNS/RTT fields and now this isn't the case anymore, but I believe our drops/DNS/RTT metrics wouldn't make sense for non-IP packets, right?

If this doesn't sound good to you, I can rollback, but I'd need to adapt also the ConvertToFLP function because I'm enforcing in my tests that the same flow input produces the exact same output regardless if it's using ConvertToFLP or converting via protobuf.

Reply:
the only thing we could be missing here is l2 drops, also pls test with DNS TCP to be sure `DNSError` is only none zero for TCP handshake I can help you with this test if u like